File: windows/src/GoalMaker.Infrastructure/Auth/SupabaseClientFactory.cs
using GoalMaker.Core.Backend;
using Supabase;

namespace GoalMaker.Infrastructure.Auth;

/// <summary>Builds the one Supabase client for the chosen backend. Called only by the composition root.</summary>
public static class SupabaseClientFactory
{
    public static Client Create(BackendEnvironment environment, string sessionPath) =>
        new(
            environment.Url,
            environment.PublishableKey,
            new SupabaseOptions
            {
                AutoRefreshToken = true,
                AutoConnectRealtime = false,
                SessionHandler = new ProtectedFileSessionPersistence(sessionPath),
            });
}

File: android/app/src/main/kotlin/com/goalmaker/app/data/supabase/SupabaseClientFactory.kt
package com.goalmaker.app.data.supabase

import com.goalmaker.app.application.environment.BackendEnvironment
import io.github.jan.supabase.SupabaseClient
import io.github.jan.supabase.auth.Auth
import io.github.jan.supabase.createSupabaseClient
import io.github.jan.supabase.realtime.Realtime

/** Builds the one Supabase client for the chosen backend. Called only by the composition root. */
object SupabaseClientFactory {
    fun create(environment: BackendEnvironment): SupabaseClient =
        createSupabaseClient(supabaseUrl = environment.url, supabaseKey = environment.publishableKey) {
            install(Auth)
            install(Realtime)
        }
}
